index.js

```diff
@@ -14,6 +14,13 @@ exports.put = function(key, value, time, timeoutCallback) {
   if (debug) {
     console.log('caching: %s = %j (@%s)', key, value, time);
   }
+
```

Comment:
It seems more logical to me to throw synchronous errors when some provides invalid types rather than try to handle the logic throughout the rest of the code. Adding the type checking here allows us to just use stuff later on without doing type checking again.